sdks/python/src/opik/evaluation/metrics/conversation/session_completeness/metric.py

```diff
@@ -0,0 +1,282 @@
+        except pydantic.ValidationError as e:
+            LOGGER.warning(
+                f"Failed to parse user goals from LLM output: {model_output}, reason: {e}",
+                exc_info=True,
+            )
+            raise e
+
+    async def _a_evaluate_user_goal(
+        self, conversation: List[Dict[str, str]], user_goal: str
+    ) -> schema.EvaluateUserGoalResponse:
+        llm_query = templates.evaluate_user_goal(
+            conversation=conversation, user_goal=user_goal
+        )
+        model_output = await self._model.agenerate_string(
+            input=llm_query, response_format=schema.EvaluateUserGoalResponse
+        )
+        try:
+            return schema.EvaluateUserGoalResponse.model_validate_json(model_output)
+        except pydantic.ValidationError as e:
+            LOGGER.warning(
+                f"Failed to parse user goal evaluation results from LLM output: {model_output}, reason: {e}",
+                exc_info=True,
+            )
+            raise e
+
+    async def _a_generate_reason(
+        self,
+        score: float,
+        verdicts: List[schema.EvaluateUserGoalResponse],
+        user_goals: List[str],
+    ) -> Optional[str]:
+        if not self._include_reason:
+            return None
+
+        negative_verdicts = _extract_negative_verdicts(verdicts)
+
+        llm_query = templates.generate_reason(
+            score=score, negative_verdicts=negative_verdicts, user_goals=user_goals
+        )
+        model_output = await self._model.agenerate_string(
+            input=llm_query, response_format=schema.ScoreReasonResponse
+        )
+        try:
+            return schema.ScoreReasonResponse.model_validate_json(model_output).reason
+        except pydantic.ValidationError as e:
+            LOGGER.warning(
+                f"Failed to parse reason from LLM output: {model_output}, reason: {e}",
+                exc_info=True,
+            )
+            raise e
+
+    async def _a_calculate_score(
+        self,
+        conversation: List[Dict[str, str]],
+    ) -> score_result.ScoreResult:
+        try:
+            user_goals = await self._a_extract_user_goals(conversation)
+            verdicts = await asyncio.gather(
+                *[
+                    self._a_evaluate_user_goal(
+                        conversation=conversation, user_goal=user_goal
+                    )
+                    for user_goal in user_goals
+                ]
+            )
+
+            score = _score_from_verdicts(verdicts=verdicts)
+            reason = await self._a_generate_reason(
+                score=score, verdicts=verdicts, user_goals=user_goals
+            )
+
+            return score_result.ScoreResult(name=self.name, value=score, reason=reason)
+        except Exception as e:
+            LOGGER.warning(
+                f"Failed to calculate session completeness quality: {e}", exc_info=True
+            )
+            return score_result.ScoreResult(
```

Comment:
Let's make the metric raise `opik.exceptions.MetricComputationError` in case there were errors inside.
When the metric is used in a standalone mode (user instantiated it and called `score` manually) we prefer to fail explicitly in case something went wrong.

However, when this metric will be used inside the evaluation loop, one of the responsibilities of the evaluation loop will be to catch MetricComputationError and construct failed ScoreResult object (we will not submit failed results to the backend, but at least the user will have access to them in their python session)